develop/data-generation/block-models.md

```diff
@@ -0,0 +1,315 @@
+  "textures": {
+    "all": "fabric-docs-reference:block/steel_block"
+  }
+}
+```
+
+It also generates a blockstate JSON file. Since we have no blockstates (e.g. Axis, Facing, ...), one variant is sufficient, and is used everytime the block is placed.
+
+```json
+{
+  "variants": {
+    "": {
+      "model": "fabric-docs-reference:block/steel_block"
+    }
+  }
+}
+```
+
+### Singletons {#singletons}
+
+The `registerSingleton` method provides JSON model files based on the `TexturedModel` you pass in and a single blockstate variant. For example
+
+@[code lang=java transcludeWith=:::datagen-model:cube-top-for-ends](@/reference/latest/src/client/java/com/example/docs/datagen/FabricDocsReferenceModelProvider.java)
+
+will generate models for a normal cube, that uses a PNG file `pipe_block` for the sides and a PNG file `pipe_block_top` for the top and bottom sides.
+
+```json
+{
+  "parent": "minecraft:block/cube_column",
+  "textures": {
+    "end": "fabric-docs-reference:block/pipe_block_top",
+    "side": "fabric-docs-reference:block/pipe_block"
+  }
+}
+```
+
+:::tip
+If you're stuck choosing which TextureModel you should use, open the `TexturedModel` class and look at the `TextureMaps`!
+:::
+
+### Block Texture Pool {#block-texture-pool}
+
+@[code lang=java transcludeWith=:::datagen-model:block-texture-pool-normal](@/reference/latest/src/client/java/com/example/docs/datagen/FabricDocsReferenceModelProvider.java)
+
+Another useful method is `registerCubeAllModelTexturePool`. Define the textures by passing in the "base block", and then append the "children", which will have the same textures. <br>
+In this case, we passed in the `RUBY_BLOCK`, so the stairs, slab and fence will use the `RUBY_BLOCK` texture.
+***It will also generate a [simple cube all JSON model](#simple-cube-all) for the "base block" to ensure the texture provision!*** <br>
+
+You can also append a `BlockFamily`, which will generate models for all of its "children".
+
+@[code lang=java transcludeWith=:::datagen-model:block-texture-pool-family](@/reference/latest/src/client/java/com/example/docs/datagen/FabricDocsReferenceModelProvider.java)
+
+
+
+::: warning
+Make sure you're using the correct Block class when passing in the block!
+For example, in the `.stairs()` method, you have to pass in an instance of StairsBlock; in `.slab()`, it has to be an instance of SlabBlock and so on! 
+:::
+
+### Doors and Trapdoors {#doors-and-trapdoors}
+
+@[code lang=java transcludeWith=:::datagen-model:door-and-trapdoor](@/reference/latest/src/client/java/com/example/docs/datagen/FabricDocsReferenceModelProvider.java)
+
+Doors and trapdoors are a little different. Here, you have to make three new textures - two for the door, and one for the trapdoor. <br>
+
+1. **The door**. It has two parts - the upper half and the lower half. **Each needs its own texture:** in this case `ruby_door_top` for the upper half and `ruby_door_bottom` for the lower.
+8 new model JSONs should be created. <br>
```

Comment:
pls clarify on the 8 new model JSONs for door? I can see what you meant by looking in the file list but not without that.

```suggestion
The `registerDoor()` method will create models for all orientations of the door, both open and closed. <br>
```